contracts/src/v0.8/dev/KeeperRegistryBase.sol

```diff
@@ -38,23 +39,41 @@ abstract contract KeeperRegistryBase is ConfirmedOwner, ExecutionPrevention, Ree
   bytes public MAX_INPUT_DATA =
     "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
 
-  address[] internal s_keeperList;
+  // Upkeep storage
   EnumerableSet.UintSet internal s_upkeepIDs;
   mapping(uint256 => Upkeep) internal s_upkeep;
-  mapping(address => KeeperInfo) internal s_keeperInfo;
-  mapping(address => address) internal s_proposedPayee;
   mapping(uint256 => bytes) internal s_checkData;
+
+  // Proposed payees for keepers
+  mapping(address => address) internal s_proposedPayee;
   mapping(address => MigrationPermission) internal s_peerRegistryMigrationPermission;
-  Storage internal s_storage;
-  uint256 internal s_fallbackGasPrice; // not in config object for gas savings
-  uint256 internal s_fallbackLinkPrice; // not in config object for gas savings
+
+  // TODO: optimise EVM word storage
+  RegistryConfig internal s_config;
+
+  // OCR config storage
+  mapping(address => Transmitter) internal s_transmitters;
+  mapping(address => Signer) internal s_signers;
+  address[] internal s_signersList; // s_signersList contains the signing address of each oracle
+  address[] internal s_transmittersList; // s_transmittersList contains the transmission address of each oracle
+  uint8 internal s_f; // Number of faulty oracles allowed
+  uint64 s_offchainConfigVersion;
+  bytes s_offchainConfig;
+
+  // OCR state
+  uint32 internal s_configCount; // incremented each time a new config is posted, The count
+  // is incorporated into the config digest to prevent replay attacks.
+  bytes32 internal s_latestConfigDigest;
+  // makes it easier for offchain systems to extract config from logs
+  uint32 internal s_latestConfigBlockNumber;
+
+  // State of registry
+  uint32 internal s_nonce; // Nonce for each upkeep created
   uint96 internal s_ownerLinkBalance;
   uint256 internal s_expectedLinkBalance;
-  address internal s_transcoder;
-  address internal s_registrar;
 
   LinkTokenInterface public immutable LINK;
-  AggregatorV3Interface public immutable LINK_ETH_FEED;
+  AggregatorV3Interface public immutable LINK_NATIVE_FEED;
   AggregatorV3Interface public immutable FAST_GAS_FEED;
```

Comment:
We should be able to get rid of `FAST_GAS_FEED`, right?

Reply:
I guess we need it for min balance getter and for the gas ceiling. Hopefully we can get rid of ceiling, since users set their own max gas price now. For the min balance, maybe we can accept a gas price as a parameter? And calculate based off that? Just some food for thought, I think the fast gas feed has been reliably unreliable and it would be cool to eliminate it if we can.

Reply:
discussed offline, will discuss removing feed dependency in future versions